chains/manager.go

```diff
@@ -393,39 +294,47 @@ func (m *manager) createChain(chainParams ChainParameters) {
 	// Note: Registering this after the chain has been tracked prevents a race
 	//       condition between the health check and adding the first chain to
 	//       the manager.
-	if chainParams.ID == constants.PlatformChainID {
+	if chain.Context.ChainID == constants.PlatformChainID {
 		if err := m.registerBootstrappedHealthChecks(); err != nil {
 			chain.Handler.StopWithError(context.TODO(), err)
 		}
 	}
 
 	// Tell the chain to start processing messages.
 	// If the X, P, or C Chain panics, do not attempt to recover
-	chain.Handler.Start(context.TODO(), !m.CriticalChains.Contains(chainParams.ID))
+	chain.Handler.Start(context.TODO(), !m.CriticalChains.Contains(chain.Context.XChainID))
```

Comment:
![image](https://github.com/ava-labs/avalanchego/assets/22109487/5b2db2d8-fc9b-4238-894a-d6dd4679df5c)
```suggestion
	chain.Handler.Start(context.TODO(), !m.CriticalChains.Contains(chain.Context.ChainID))
```

Reply:
Yeah this is a bug